plugins/gcp/resoto_plugin_gcp/collector.py

```diff
@@ -120,11 +121,11 @@ def rm_nodes(cls, ignore_kinds: Optional[Type[Any]] = None) -> None:
             remove_nodes.clear()
 
         # nodes need to be removed in the correct order
-        rm_nodes((compute.GcpNodeType, compute.GcpDiskType))
-        rm_nodes(compute.GcpMachineType, compute.GcpAcceleratorType)  # ignore accelerator types
-        rm_nodes(compute.GcpAcceleratorType)
-        rm_nodes(billing.GcpSku)
-        rm_nodes(billing.GcpService)
+        for resource_class in resources_to_remove:
+            if resource_class == compute.GcpMachineType:
+                rm_nodes(compute.GcpMachineType, compute.GcpAcceleratorType)  # ignore accelerator types
```

Comment:
TBH: I don't think spreading the knowledge about the deletion of resources is better than what we have.